### ISSUE TITLE
TypeError: data type 'category' not understood

### ISSUE THREAD

**Issue by Metaming** [JoaquinAmatRodrigo/skforecast#446]
I follow  the examples 
https://skforecast.org/0.5.0/user_guides/autoregresive-forecaster-exogenous.html
and  https://skforecast.org/0.5.0/user_guides/prediction-intervals.html

to build multi-step quantile forecasters.

The codes are like this:
forecaster_q50 = ForecasterAutoregDirect(
                     regressor = LGBMRegressor(
                                     objective     = 'quantile',
                                     metric        = 'quantile',
                                     alpha         = 0.5,
                                     learning_rate = 0.1,
                                     max_depth     = 4,
                                     n_estimators  = 100
                                 ),
                     lags = 7*48,
                     steps = 48
                 )
forecaster_q50.fit(y=df['power'],
                              exog = df[['interval','day_of_week','month','id']]) 

------------------------------------------------------------
issue: when trying to use exog variables 

the following error occurs

TypeError                                 Traceback (most recent call last)
<ipython-input-26-606eab1f03b1> in <module>
      1 tic = time.time()
      2 
----> 3 forecaster_q50.fit(y=time_df['power'],
      4                   exog = time_df[['interval','day_of_week','month']]) 
      5 

~\AppData\Roaming\Python\Python38\site-packages\skforecast\ForecasterAutoregDirect\ForecasterAutoregDirect.py in fit(self, y, exog)
    626                  exog.columns.to_list() if isinstance(exog, pd.DataFrame) else exog.name
    627 
--> 628         X_train, y_train = self.create_train_X_y(y=y, exog=exog)
    629 
    630         # Train one regressor for each step

~\AppData\Roaming\Python\Python38\site-packages\skforecast\ForecasterAutoregDirect\ForecasterAutoregDirect.py in create_train_X_y(self, y, exog)
    437 
    438             check_exog(exog=exog, allow_nan=False)
--> 439             check_exog_dtypes(exog)
    440             self.exog_dtypes = get_exog_dtypes(exog=exog)
    441 

~\AppData\Roaming\Python\Python38\site-packages\skforecast\utils\utils.py in check_exog_dtypes(exog)
    343 
    344     if isinstance(exog, pd.DataFrame):
--> 345         if not all([dtype in ['float', 'int', 'category'] for dtype in exog.dtypes]):
    346             warnings.warn(
    347                 ("`exog` may contain only `int`, `float` or `category` dtypes. Most "

~\AppData\Roaming\Python\Python38\site-packages\skforecast\utils\utils.py in <listcomp>(.0)
    343 
    344     if isinstance(exog, pd.DataFrame):
--> 345         if not all([dtype in ['float', 'int', 'category'] for dtype in exog.dtypes]):
    346             warnings.warn(
    347                 ("`exog` may contain only `int`, `float` or `category` dtypes. Most "

TypeError: data type 'category' not understood

-----------------------------------------------

The error is not related to the use of category dtype in the dataframe, but the internal process of dtype checking.

I looked into the source code in .\skforecast\utils\utils.py and the following line which caused the error
--> 345         if not all([dtype in ['float', 'int', 'category'] for dtype in exog.dtypes]):

I believe it is due to the fact that dtype is not a 'str' but rather a pandas.CategoricalDType, so if you are trying to compare it with a string in the list, it gives error. If you want to compare the name of the dtype explicitly, you can change the line to

if not all([dtype.name in ['int8','int16','int32','int64','float16','float32','float64','category']  for dtype in exog.dtypes]):

or 

if not all ([any(type_allowed in dtype.name for type_allowed in ['float','int','category']) for dtype in exog.dtypes ]):

-----------------------------------------------------
Hope it helps

**Comment by JavierEscobarOrtiz** [comment]
Hello @Metaming, 

Thank you for opening the issue. Could you please add a more complete example of your code including the regressor you are using?

**Comment by Metaming** [comment]
> Hello @Metaming,
> 
> Thank you for opening the issue. Could you please add a more complete example of your code including the regressor you are using?

Hi @JavierEscobarOrtiz , please see the updated post. Cheers

**Comment by JavierEscobarOrtiz** [comment]
Thanks @Metaming, are you using skforecast 0.5.0? In that version, we didn't include support for categorical variables.

If that is the case, I recommend you use skforecast 0.8.0 and read this user guide:

https://skforecast.org/latest/user_guides/categorical-features.html

If you are using skforecast 0.8.0 just let us know if you find this problem 😄

**Comment by Metaming** [comment]
> Hi @Metaming ,
> 
> Thank you very much for reporting this bug! I really appreciate it.
> 
> It appears that the code `df["col1"].dtype == "category" ` only functions correctly in pandas versions 1.4 and above.
> 
> I will incorporate your suggestion of comparing the dtype names instead. In the meantime, would you mind updating pandas and checking if the error persists?
> 
> Thank you!

Hi @JoaquinAmatRodrigo and @JavierEscobarOrtiz , thanks for incorporate my suggestion. I am using pandas 1.5.1, and my skforecast is the latest version 0.8.0

**Comment by JoaquinAmatRodrigo** [comment]
Interesting,
Could you try this code to better understand where the issue comes from?

```
import pandas as pd

df = pd.DataFrame({'col1': [1, 2, 3], 'col2': ["a", "b", "c"]}) 
df['col2'] = df['col2'].astype('category')

print(df['col2'].dtype == 'category')
print(pd.__version__)
```

The output I get is 

```
True
1.5.3
```

**Comment by Metaming** [comment]
> [dtype in ['float', 'int', 'category'] for dtype in exog.dtypes]

I got the same result, except the version is 1.5.1.

But maybe you can try 

print([dtype == 'category' for dtype in df[['col1','col2']].dtypes])

and see if it returns error.

**Comment by JavierEscobarOrtiz** [comment]
Hi @Metaming,

We have fixed the bug in skforecast 0.8.1. You can see your contribution in the [realese file](https://github.com/JoaquinAmatRodrigo/skforecast/blob/master/docs/releases/releases.md).

Thank you very much!
